src/views/Home.js

```diff
@@ -22,13 +39,25 @@ export default class Home extends Component {
 Home.propTypes = {
     navigation: PropTypes.shape({
         navigate: PropTypes.func
-    })
+    }),
+    setLocation: PropTypes.func.isRequired
 }
 
 Home.navigationOptions = {
     title: 'Home'
 };
 
+const mapStateToProps = () => ({});
+
+const mapDispatchToProps = dispatch => ({
+    setLocation: loc => dispatch(setLocation(loc))
```

Comment:
Still isn't totally clear to me how `Home` has access to `setLocation`. I guess its through `connect` - I almost get it 🤔

Reply:
Exactly, `connect` bolts on redux stuff to a component during the render cycle by executing the `mapDispatchToProps` and `mapStateToProps` functions. Those functions return objects, and they are merged with the props before being passed to the component.